src/primaite/agents/llm/agent.py

```diff
@@ -45,67 +52,182 @@ class LLM:
     def __init__(self, base_url: str, timeout: int = 60) -> None:
         self.client = Client(base_url=base_url, timeout=timeout)
 
-    def predict(self, env_state: EnvironmentState, env_history: list[EnvironmentState]):
-        # Task explanation
-        system_msg = "You are a cyber-defensive agent. Your mission is to protect a network against red agent attacks. You have a limited view of the network environment, known as an observation space. The network is made up of nodes (e.g. computers, switches) and links between some of the nodes, through which information is transmitted using standard protocols. The current link traffic is displayed as a percentage of its total bandwidth, for each possible protocol. The red agent is trying to compromise the network, by either directly incapacitating the nodes HARDWARE, SOFTWARE or FILE_SYSTEM states, or indirectly overwhelming the nodes through denial-of-service type attacks. \n"
-        env = env_state.env
-        # Action space description
-        action_space_desc = get_action_space_description(env_state.env)
-        system_msg += action_space_desc
+    def generate(
+        self, prompt: str, repetition_penalty: Optional[float] = None, max_new_tokens: Optional[int] = 1024
```

Comment:
i dont think max_new_tokens should be optional

Reply:
Just means that you don't have to pass it and it will just default

Reply:
if you wrap it the type in an optional it allows it to be potentially None but that gives me type hint errors because the tgi client generate method can't accept None for max new tokens

Reply:
Yeah you're right, my bad didn't have type checking on